README.md

```diff
@@ -14,169 +12,202 @@
+- Run OLAP queries from any SQL client both on the full data set as well as sampled data (showcasing sub-second
+interactive query speeds). The stratified sample allows us to manage an infinitely growing data set at a fraction of the
+cost otherwise required.
 
 ### Ad Impression Analytics use case
-We borrow our use case implementation from this [blog](https://chimpler.wordpress.com/2014/07/01/implementing-a-real-time-data-pipeline-with-spark-streaming/) - We more or less use the same data structure and aggregation logic and we have adapted this code to showcase the SnappyData programming model extensions to Spark. We retain the native Spark example for comparison.
+We borrow our use case implementation from this [blog](https://chimpler.wordpress.com/2014/07/01/implementing-a-real-time-data-pipeline-with-spark-streaming/)
+\- We more or less use the same data structure and aggregation logic and we have adapted this code to showcase the
+SnappyData programming model extensions to Spark. We have also kept the native Spark example using structured streaming
+for comparison.
 
 Our architecture is depicted in the figure below.
 
-We consider an adnetwork where adservers log impressions in [Apache Kafka](http://kafka.apache.org/) (distributed publish-subscribe messaging system). These impressions are then aggregated by [Spark Streaming](http://spark.apache.org/streaming/) into the SnappyData Store. External clients connect to the same cluster using JDBC/ODBC and run arbitrary OLAP queries.
-As AdServers can feed logs from many websites and given that each AdImpression log message represents a single Ad viewed by a user, one can expect thousands of messages every second. It is crucial that ingestion logic keeps up with the stream. To accomplish this, SnappyData collocates the store partitions with partitions created by Spark Streaming. i.e. a batch of data from the stream in each Spark executor is transformed into a compressed column batch and stored in the same JVM, avoiding redundant shuffles (except for HA).
+We consider an adnetwork where adservers log impressions in [Apache Kafka](http://kafka.apache.org/). These impressions
+are then aggregated using [Spark Structured Streaming](https://spark.apache.org/docs/latest/structured-streaming-programming-guide.html) into the SnappyData Store. External clients connect to the same cluster using JDBC/ODBC and run arbitrary OLAP queries.
+As AdServers can feed logs from many websites and given that each AdImpression log message represents a single Ad viewed
+by a user, one can expect thousands of messages every second. It is crucial that ingestion logic keeps up with the
+stream. To accomplish this, SnappyData collocates the store partitions with partitions created by Spark Streaming.
```

Comment:
Does the collocation happen automatically? In any case, would be better to mention briefly how does it happen.

Reply:
Won't be easy to explain collocation in brief. Will see whether we have any link in documentation which we can provide here.